lightning_pose_app/backend/project.py

```diff
@@ -0,0 +1,266 @@
+import os
+import json
+import h5py
+import pandas as pd
+import numpy as np
+from PIL import Image
+import io
+
+from lightning_pose_app import (
+    COLLECTED_DATA_FILENAME,
+    LABELED_DATA_DIR,
+    VIDEOS_DIR,
+)
+
+
+def extract_frames_from_pkg_slp(file_path, base_output_dir):
+    with h5py.File(file_path, 'r') as hdf_file:
+        # Identify video names
+        video_names = {}
+        for video_group_name in hdf_file.keys():
+            if video_group_name.startswith('video'):
+                source_video_path = f'{video_group_name}/source_video'
+                if source_video_path in hdf_file:
+                    source_video_json = hdf_file[source_video_path].attrs['json']
+                    source_video_dict = json.loads(source_video_json)
+                    video_filename = source_video_dict['backend']['filename']
+                    video_names[video_group_name] = video_filename
+
+        # Extract and save images for each video
+        for video_group, video_filename in video_names.items():
+            output_dir = os.path.join(
+                base_output_dir, LABELED_DATA_DIR, os.path.basename(video_filename).split('.')[0]
+            )
+            if not os.path.exists(output_dir):
+                os.makedirs(output_dir)
+
+            if video_group in hdf_file and 'video' in hdf_file[video_group]:
+                video_data = hdf_file[f'{video_group}/video'][:]
+                frame_numbers = hdf_file[f'{video_group}/frame_numbers'][:]
+                frame_names = []
+                for i, (img_bytes, frame_number) in enumerate(zip(video_data, frame_numbers)):
+                    img = Image.open(io.BytesIO(np.array(img_bytes, dtype=np.uint8)))
+                    frame_name = f"img{str(frame_number).zfill(8)}.png"
+                    img.save(f"{output_dir}/{frame_name}")
+                    frame_names.append(frame_name)
+                    print(f"Saved frame {frame_number} as {frame_name}")
+
+
+# Function to extract data and create the required DataFrame for multiple videos
+def extract_labels_from_pkg_slp(file_path):
+    data_frames = []
+    scorer_row, bodyparts_row, coords_row = None, None, None
+
+    with h5py.File(file_path, 'r') as hdf_file:
```

Comment:
since this block is used twice, you should make a function `extract_video_names_from_pkg_slp` that returns a list of videos, then use it in this function and the one above